packages/verified-fetch/src/utils/handle-redirects.ts

```diff
@@ -0,0 +1,91 @@
+import { type AbortOptions, type ComponentLogger } from '@libp2p/interface'
+import { type VerifiedFetchInit, type Resource } from '../index.js'
+import { matchURLString } from './parse-url-string.js'
+import { movedPermanentlyResponse } from './responses.js'
+import type { CID } from 'multiformats/cid'
+
+interface GetRedirectResponse {
+  cid: CID
+  resource: Resource
+  options?: Omit<VerifiedFetchInit, 'signal'> & AbortOptions
+  logger: ComponentLogger
+
+}
+
+function maybeAddTraillingSlash (path: string): string {
+  // if it has an extension-like ending, don't add a trailing slash
+  if (path.match(/\.[a-zA-Z0-9]{1,4}$/) != null) {
+    return path
+  }
+  return path.endsWith('/') ? path : `${path}/`
+}
+
+export async function getRedirectResponse ({ resource, options, logger, cid }: GetRedirectResponse): Promise<null | Response> {
```

Comment:
What do you think about adding a test for this function that illustrate when it redirects?
```suggestion
// See https://specs.ipfs.tech/http-gateways/path-gateway/#location-response-header
export async function getRedirectResponse ({ resource, options, logger, cid }: GetRedirectResponse): Promise<null | Response> {
```

Reply:
It really should have a test, but gwc tests it, so i figured it was fine for now. Since we already have gwc tests, redoing all that coverage in unit tests is not ideal.

Ill add a basic test